plugins/module_utils/foreman_helper.py

```diff
@@ -220,25 +216,26 @@ def __init__(self, **kwargs):
             password=dict(required=True, no_log=True),
             validate_certs=dict(type='bool', default=True, aliases=['verify_ssl']),
         )
-        argument_spec.update(kwargs.pop('argument_spec', {}))
         argument_spec.update(gen_args)
+        argument_spec.update(kwargs.pop('argument_spec', {}))
         supports_check_mode = kwargs.pop('supports_check_mode', True)
-        self._aliases = {alias for arg in argument_spec.values() for alias in arg.get('aliases', [])}
 
         self.required_plugins = kwargs.pop('required_plugins', [])
 
         super(ForemanAnsibleModule, self).__init__(argument_spec=argument_spec, supports_check_mode=supports_check_mode, **kwargs)
 
-        self._params = self.params.copy()
+        if 'verify_ssl' in self.params:
+            self.warn("Please use 'validate_certs' instead of deprecated 'verify_ssl'.")
+
+        aliases = {alias for arg in argument_spec.values() for alias in arg.get('aliases', [])}
```

Comment:
TIL: you can have multiple `for` in a list comprehension 🤯

Reply:
The order of the expressions is unintuitive though.